File: VBAudioRouter/Utils/SliderTimeSpanValueConverter.vb
Namespace Utils

    Public Class SliderTimeSpanValueConverter
        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
            If targetType = GetType(String) Then
                Return TimeSpan.FromMilliseconds(CType(value, Double)).ToString("mm\:ss")
            ElseIf value.GetType() = GetType(TimeSpan) Then
                Return DirectCast(value, TimeSpan).TotalMilliseconds
            Else
                Throw New NotImplementedException()
            End If
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
            Return Convert(value, targetType, parameter, language)
        End Function
    End Class

End Namespace

File: VBAudioRouter/Utils/SliderTimeSpanValueConverter.cs
using Microsoft.UI.Xaml.Data;

namespace VBAudioRouter.Utils;

internal sealed class SliderTimeSpanValueConverter : IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, string language)
    {
        if (targetType == typeof(string))
            return TimeSpan.FromMilliseconds(System.Convert.ToDouble(value)).ToString(@"mm\:ss");

        if (value is TimeSpan timespan)
            return timespan.TotalMilliseconds;

        throw new NotImplementedException();
    }

    public object ConvertBack(object value, Type targetType, object parameter, string language)
        => Convert(value, targetType, parameter, language);
}
